router.php

```diff
@@ -49,12 +50,6 @@
     if (false === $target) {
         throw new RuntimeException('FUNCTION_TARGET is not set');
     }
-    if (!is_callable($target)) {
-        throw new InvalidArgumentException(sprintf(
-            'Function target is not callable: "%s"',
-            $target
-        ));
-    }
 
     $signatureType = getenv('FUNCTION_SIGNATURE_TYPE', true) ?: 'http';
```

Comment:
I think a better design (to minimize use of the global scope / singletons, and to make use of better design patterns such as dependency injection) the following api should be used instead (see my suggested changes to `FunctionsFramework` to see how this is implemented):
```php
$registry = FunctionsFramework::registry();
$invoker = new Invoker($target, $signatureType, $registry);
```

Reply:
Hmm, I'm not sure we want to expose the registry on the `FunctionsFramework` class. This is the only class that PHP function writers **must** use, and the function registry is really an implementation detail. We want to discourage users from accessing the registry directly as that may cause unintended behavior.

Reply:
Either way we are exposing the registry to the users. The current implementation allows them to access the singleton with `Registry::getInstance`. There is no functional difference between this and `FunctionsFramework::registry`. The main conceptual difference is we are reducing the number of singleton classes to one instead of two, and making it clear that the `Registry` is being managed by the Functions Framework, and is only a singleton in the context of the Functions Framework.

Reply:
How about we put the static registry instance on the `Invoker` class, and have `FunctionsFramework` use `Invoker::registry()`?

I think there's a significant user experience difference when writing a function because we expect `use Google\CloudFunctions\FunctionsFramework;` to be included, so autocomplete will show anything on that class, but not on `Invoker`. It's unlikely that function authors would use `Invoker.php` directly.